openapi-contracts/src/main/com/sinch/sdk/domains/voice/models/dto/v1/WebhooksEventDto.java

```diff
@@ -0,0 +1,516 @@
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.JsonToken;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.MapperFeature;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import com.fasterxml.jackson.databind.ser.std.StdSerializer;
+import com.sinch.sdk.core.models.AbstractOpenApiSchema;
+import com.sinch.sdk.core.utils.databind.JSONNavigator;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+@JsonDeserialize(using = WebhooksEventDto.WebhooksEventDtoDeserializer.class)
+@JsonSerialize(using = WebhooksEventDto.WebhooksEventDtoSerializer.class)
+public class WebhooksEventDto extends AbstractOpenApiSchema {
+  private static final Logger log = Logger.getLogger(WebhooksEventDto.class.getName());
+
+  public static class WebhooksEventDtoSerializer extends StdSerializer<WebhooksEventDto> {
+    public WebhooksEventDtoSerializer(Class<WebhooksEventDto> t) {
+      super(t);
+    }
+
+    public WebhooksEventDtoSerializer() {
+      this(null);
+    }
+
+    @Override
+    public void serialize(WebhooksEventDto value, JsonGenerator jgen, SerializerProvider provider)
+        throws IOException, JsonProcessingException {
+      jgen.writeObject(value.getActualInstance());
+    }
+  }
+
+  public static class WebhooksEventDtoDeserializer extends StdDeserializer<WebhooksEventDto> {
+    public WebhooksEventDtoDeserializer() {
+      this(WebhooksEventDto.class);
+    }
+
+    public WebhooksEventDtoDeserializer(Class<?> vc) {
+      super(vc);
+    }
+
+    @Override
+    public WebhooksEventDto deserialize(JsonParser jp, DeserializationContext ctxt)
+        throws IOException, JsonProcessingException {
+      JsonNode tree = jp.readValueAsTree();
+      Object deserialized = null;
+      WebhooksEventDto newWebhooksEventDto = new WebhooksEventDto();
+      Map<String, Object> result2 =
+          tree.traverse(jp.getCodec()).readValueAs(new TypeReference<Map<String, Object>>() {});
+      String discriminatorValue = (String) result2.get("event");
+      switch (discriminatorValue) {
+        case "ace":
+          deserialized = tree.traverse(jp.getCodec()).readValueAs(AceRequestDto.class);
+          newWebhooksEventDto.setActualInstance(deserialized);
+          return newWebhooksEventDto;
+        case "aceRequest":
```

Comment:
What is the difference between `ace` and `aceRequest` ?

Reply:
None.
This is coming from generator tool (WebhooksEventDto.java is a generated file)
When a `oneOf` is defined from OAS it create a switch for related schema name

According to following OAS description:
```
    webhooksEvent:
      oneOf:
      ...
        - $ref: "#/components/schemas/aceRequest"
      ...
      discriminator:
        propertyName: event
        mapping:
         ...
          ace: "#/components/schemas/aceRequest"
         ...
``` 
The ` - $ref: "#/components/schemas/aceRequest"` generate the `case aceRequest` to be able to parse a JSON ... but we are not using it because of the `mapping` definition.
There was a discussion onto openapi generator project related to this point (sorry I do not have link) to remove it from generation; and to be honest I did not investigate (yet) more onto this point